### PR TITLE
Re-enable fragment driver

### PR DESCRIPTION
**Pull request by ivanorman**
Fragment driver re-enabled, testing updated.

### CODE REVIEW

File: tests/cxx/unit_tests/ghostfragment/drivers/fragment.cpp [hunk 1]
```diff
@@ -1,78 +1,64 @@
...
-//         return rv;
-//     });
-// }
+/* Testing strategy:
+ *
+ * The FragmentDriver module is purely a driver. If we assume the modules it
+ * calls work correctly, the only thing we need to test is that the data flows.
+ * Testing is done using the default modules (Heavy Atom for pseudoatom generation,
+ * Bond-Based for fragmentation).
+ */
```
Comment (ryanmrichard) [inline tests/cxx/unit_tests/ghostfragment/drivers/fragment.cpp:23]: For an integration test that's okay (so if you want to create a directory `tests/cxx/integration_tests/ghostfragment/drivers` and move this file there that's fine), but for unit tests we really want to be testing this module in isolation. This in turn means we need to re-enable the lambda modules you commented out.

To clarify, unit tests try to test a code component in relative isolation. Meaning we want the test to have minimal additional dependencies. The goal of a unit test is really to try and catch any sort of problem intrinsic with the code we are testing. 

Integration tests couple several pieces together and make sure they work smoothly together. Unlike unit tests, integration tests can fail if any of the pieces break; however, if one has unit tests for each piece, it's also likely that the unit test for that piece is also failing. Ultimately, the goal of integration tests are to make sure there's no surprises when you start putting the pieces together.

Comment (ivanorman) [reply tests/cxx/unit_tests/ghostfragment/drivers/fragment.cpp:47]: Added integration test, changed unit test to use lambda modules.

File: tests/cxx/integration_tests/ghostfragment/drivers/fragment.cpp
Comment (ryanmrichard) [inline tests/cxx/integration_tests/ghostfragment/drivers/fragment.cpp:1]: These tests aren't being run. You need to add:

```.cmake
    cpp_add_tests(
        "integration_tests_for_${PROJECT_NAME}"
        SOURCE_DIR "${cxx_tests_dir}/integration_tests/${PROJECT_NAME}"
        INCLUDE_DIR "${project_src_dir}"
        DEPENDS Catch2::Catch2 ${PROJECT_NAME}
    )
```
to the `BUILD_TESTING` block of `CMakeLists.txt` (lines 69-93).

Comment (ryanmrichard) [reply tests/cxx/integration_tests/ghostfragment/drivers/fragment.cpp:1]: Were you going to add the integration tests back in, or did you decide to just skip them?

Comment (ivanorman) [reply tests/cxx/integration_tests/ghostfragment/drivers/fragment.cpp:1]: I elected to just skip them.

**Pull request merged**